pkg/controller/statusmanager/status_manager.go

```diff
@@ -184,6 +185,29 @@ func (status *StatusManager) SetDeployments(deployments []types.NamespacedName)
 func (status *StatusManager) SetRelatedObjects(relatedObjects []configv1.ObjectReference) {
 	status.Lock()
 	defer status.Unlock()
+
+	var found bool = false
+	for _, status_obj := range status.relatedObjects {
+		for _, obj := range relatedObjects {
+			found = reflect.DeepEqual(status_obj, obj)
+
+			if found {
+				break
+			}
+		}
+		if !found {
+			obj_to_delete := &uns.Unstructured{}
+			key := types.NamespacedName{
+				Name:      status_obj.Name,
+				Namespace: status_obj.Namespace,
+			}
+			err := status.client.Get(context.TODO(), key, obj_to_delete)
```

Comment:
This won't work; the object you pass in has to be a proper API object, so the client can introspect its type and figure out what kind of object to get. (Just specifying the name isn't enough, because eg, it's totally valid to have a Service and Pod with the same name in the same namespace.) You should be able to find the right golang type with the `RESTMapper`... it's basically the opposite of the operation you perform in `operconfig_controller.go` to fill in `status_obj.Group` and `.Resource`. You might need to use something in `reflect` to actually construct an empty object of that type. (It might make sense at the same time to change `StatusManager.SetRelatedObjects` to take a `[]interface{}` of API objects and construct the array of `ObjectReference`s itself. That way all of the `RESTMapper` code is in `StatusManager` rather than being split between there and the operator config controller.)